### DISCUSSION TITLE
Cognito unauthorized_client

### DISCUSSION THREAD

**Discussion by hauerm** [nextauthjs/next-auth#6999] [Help]
Hello

I keep receiving unauthorized_client / TODO: Handle OIDC response body error response when running authjs (0.5.1, sveltekit 0.3.0) from in a Lambda behind Cloudfront and Api-Gateway together with Cognito. When I run from localhost everything works fine.

Custom domain setup:
* Cloudfront main entry: app.example.com
* Cognito: auth.app.example.com
* Api-Gateway: api.app.example.com

![image](https://user-images.githubusercontent.com/89738886/226177591-a3094097-6202-4f14-9a96-10707099acc0.png)

I set the following env variables:
![image](https://user-images.githubusercontent.com/89738886/226443177-ad2b1bdd-8ceb-4e6a-b62d-3e62ece37c7e.png)

Maybe someone can give me a hint on what I am missing. I pass all cookies and headers, this seems to look ok when I investigate the requests/responses.

**Comment by steve-scorn** [comment]
@hauerm - did you ever get a solution to this? I have exactly the same error with Cognito. Works on localhost, but not when deployed to Amplify Hosting.

**Comment by drelit** [comment]
Following up on this issue. Any resolution?
